### PR TITLE
Look for human-readable names for conditions

### PR DESCRIPTION
**Pull request by JNygaard-Skylight**
# PULL REQUEST

## Summary

The message-parser config the seed script uses is not getting human readable names.

## Related Issue

Fixes # n/a

## Acceptance Criteria

Please copy the acceptance criteria from your ticket and paste it here for your reviewer(s)

## Additional Information

Anything else the review team should know?

## Checklist

- [ ] If this code affects the other scrum team, have they been notified? (In Slack, as reviewers, etc.)

[//]: # "PR title: Remember to name your PR descriptively!"

### CODE REVIEW

File: containers/message-parser/app/default_schemas/extended.json [hunk 1]
```diff
@@ -349,7 +349,7 @@
         }
       },
       "condition": {
-        "fhir_path": "Observation.valueCodeableConcept.coding.display",
+        "fhir_path": "iif(Observation.valueCodeableConcept.text.exists(), Observation.valueCodeableConcept.text, Observation.valueCodeableConcept.coding.display)",
```
Comment (mcmcgrath13) [inline containers/message-parser/app/default_schemas/extended.json:352]: does the display need a `first` since there can be multiple?

File: containers/trigger-code-reference/app/utils.py [hunk 2]
```diff
@@ -205,53 +205,89 @@ def _find_codes_by_resource_type(resource: dict) -> List[str]:
...
-            and x["code"] == "64572001"  # Condition
-        ]:
-            condition_name = _get_condition_name_from_snomed_code(code)
-            if condition_name:
-                resource["valueCodeableConcept"]["text"] = condition_name
+        ),
+        None,
+    )
+
+    if condition_code:
```
Comment (mcmcgrath13) [inline containers/trigger-code-reference/app/utils.py:259]:
```suggestion
    if condition_code is not None:
```

File: containers/trigger-code-reference/app/utils.py [hunk 3]
```diff
@@ -205,53 +205,89 @@ def _find_codes_by_resource_type(resource: dict) -> List[str]:
...
+        human_readable_condition_name = _get_condition_name_from_snomed_code(
+            condition_code["code"]
+        )
+
+        if human_readable_condition_name:
+            resource["valueCodeableConcept"]["text"] = human_readable_condition_name
+        elif "display" in condition_code:
+            resource["valueCodeableConcept"]["text"] = condition_code["display"]
```
Comment (mcmcgrath13) [inline containers/trigger-code-reference/app/utils.py:267]: what's the thinking around including the `display` as a fallback? the display would be in the resulting fhir bundle either way, right? so it's not stamping new/outside info

File: containers/trigger-code-reference/seed-scripts/seed_rckms_database.py [hunk 4]
```diff
@@ -3,174 +3,32 @@
 seed_rckms_database.py
 ===========
 
-A significantly paired down version of the original script to seed the RCKMS database. This script will extract the conditions from the RCKMS docx files and load them into the SQLite database.
-
-
-Example:
-    $ python seed_rckms_database.py <archive.zip> <database.db>
+A significantly paired down version of the original script to seed the RCKMS database. This script will extract the conditions from the RCKMS CSV in assets (downloaded from https://www.rckms.org/content-repository) files and load them into the SQLite database.
```
Comment (mcmcgrath13) [inline containers/trigger-code-reference/seed-scripts/seed_rckms_database.py:6]:
```suggestion
A significantly pared down version of the original script to seed the RCKMS database. This script will extract the conditions from the RCKMS CSV in assets (downloaded from https://www.rckms.org/content-repository) files and load them into the SQLite database.
```

File: containers/trigger-code-reference/seed-scripts/seed_rckms_database.py [hunk 5]
```diff
@@ -3,174 +3,32 @@
...
-        for condition in conditions:
-            # insert the Condition into the database if it does not exist
+        next(reader)
+        for row in tqdm(reader, desc="Loading conditions", unit=" conditions"):
             cur.execute(
-                "INSERT INTO conditions (id, system, name) "
-                "VALUES (:snomed, 'http://snomed.info/sct', :name) "
-                "ON CONFLICT DO NOTHING",
-                condition.__dict__,
+                "INSERT INTO conditions (id, system, name, description) VALUES (?, ?, ?, ?)",
```
Comment (mcmcgrath13) [inline containers/trigger-code-reference/seed-scripts/seed_rckms_database.py:31]: do we ever get conflicts in practice?

File: containers/message-parser/app/default_schemas/extended.json (hunk 1 above)
Comment (JNygaard-Skylight) [reply containers/message-parser/app/default_schemas/extended.json:352]: No, `evaluate` always returns a list, plus the logic to pick a `display` is easier in TS.

File: containers/trigger-code-reference/app/utils.py (hunk 2 above)
Comment (JNygaard-Skylight) [reply containers/trigger-code-reference/app/utils.py:259]: I don't know why we need to test specifically for `is not None` instead of is `condition_code` truthy.

File: containers/trigger-code-reference/app/utils.py (hunk 3 above)
Comment (JNygaard-Skylight) [reply containers/trigger-code-reference/app/utils.py:267]: I was thinking that it might make the frontend logic easier if there was always a `"text"` even if it is not the preferred human-readable name. But perhaps it might be confusing if we expect `"text"` to be the preferred name and instead it's just the same as the `"display"`?

File: containers/trigger-code-reference/seed-scripts/seed_rckms_database.py (hunk 5 above)
Comment (JNygaard-Skylight) [reply containers/trigger-code-reference/seed-scripts/seed_rckms_database.py:31]: I don't think there is any reason why we should. You can check the CSV: `containers/trigger-code-reference/assets/RCKMS Condition Codes.csv` and if we did I'm okay with it blowing up.
